run/pubsub/main.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START run_pubsub_server]
+
+// Sample run-pubsub is a Cloud Run service which handles Pub/Sub messages.
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	http.HandleFunc("/", HelloPubSub)
+	// Determine port for HTTP service.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+	// Start HTTP server.
+	log.Printf("Listening on port %s", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+}
+
+// [END run_pubsub_server]
+
+// [START run_pubsub_handler]
+// PubSubMessage is the payload of a Pub/Sub event. Please refer to the docs for
+// additional information regarding Pub/Sub events.
```

Comment:
Link to https://cloud.google.com/functions/docs/tutorials/pubsub, even though it's about Cloud Functions?

Reply:
Why not link to cloud.google.com/run/docs/tutorials/pubsub? The reason this is vague is that the specific docs pages to look at for a definition of event structure are unclear and in flux, but there's probably something closer in the Pub/Sub docs. Do you have a preference?